js/app/controllers/playercontroller.js

```diff
@@ -479,10 +493,61 @@ function ($scope, $rootScope, playlistService, Audio, gettextCatalog, Restangula
 		}
 	};
 
+	// Seekbar preview mouse support
+	$scope.seekbarPreview = function($event) {
+		if (!$scope.player.seekingSupported()) return;
+
+		var offsetX = $event.offsetX || $event.originalEvent.layerX;
+		var ratio = offsetX / $event.currentTarget.clientWidth;
+		var timestamp = ratio * $scope.position.total;
+		
+		$scope.position.currentPreview = timestamp;
+		$scope.position.currentPreview_ts = Date.now();
+	};
+
+	$scope.seekbarEnter = function() {
+		// Simple filter for transient mouse movements
+		$scope.position.currentPreview_tf = setTimeout(function() {
+			// Force AngularJS to refresh/render scope
+			$scope.$apply(function() {
+				$scope.position.currentPreview_tf = null;
+			});			
+		}, 100);
+	};
+
+	$scope.seekbarLeave = function() {
+		$scope.position.currentPreview = null;
+		$scope.position.currentPreview_ts = null;
+	};
+
+	// Seekbar preview touch support
+	$scope.seekbarTouchLeave = function($event) {
+		if (!$scope.player.seekingSupported() || $event?.type !== 'touchend') return;
+		
```

Comment:
I usually try to avoid multiple exit points in the functions, unless the additional exit point shortens or simplifies the function significantly. Here, inverting the if condition and having the logic within the body of the if wouldn't be any more complex:
```
if ($scope.player.seekingSupported() && $event?.type === 'touchend') {
    // do the calculation and seeking
}
```

Reply:
I like to keep the functions as lean as possible. I worked on a big project from the callback **hell** once and have refactored the whole code base.. since then I try to avoid nesting and if something is considered as abort condition, I will do it fast and lean on the beginning of the function.

But no problem, I will change the corresponding locations.